examples/car_fsm.erl

```diff
@@ -0,0 +1,543 @@
+   || Speed < ?SLOW_THRESHOLD] ++
+    [{normal, ?CALL(accelerate, [accelerator(Speed, normal)])}
+     || Speed < ?NORMAL_THRESHOLD] ++
+    [{fast, ?CALL(accelerate, [accelerator(Speed, fast)])}
+     || Speed < ?FAST_THRESHOLD] ++
+    [{too_fast, ?CALL(accelerate, [accelerator(Speed, too_fast)])}
+     || Speed < ?MAX_SPEED].
+
+brake_commands(Speed) ->
+  [{stopped, ?CALL(brake, [braker(Speed, stopped)])}] ++
+    [{slow, ?CALL(brake, [braker(Speed, slow)])} || Speed > 1] ++
+    [{normal, ?CALL(brake, [braker(Speed, normal)])}
+     || Speed > ?SLOW_THRESHOLD + 1] ++
+    [{fast, ?CALL(brake, [braker(Speed, fast)])}
+     || Speed > ?NORMAL_THRESHOLD + 1] ++
+    [{too_fast, ?CALL(brake, [braker(Speed, too_fast)])}
+     || Speed > ?FAST_THRESHOLD + 1].
+
+
+%% -----------------------------------------------------------------------------
+%% Properties
+%% -----------------------------------------------------------------------------
+
+
+%% This should "never" fail:<br/>
+%% `proper:quickcheck(car_fsm:prop_normal_distance(), 1000).'
+prop_distance() ->
+  ?FORALL(Cmds, proper_fsm:commands(?MODULE),
+          ?TRAPEXIT(
+             begin
+               start_link(),
+               {_H, {_, S}, R} = proper_fsm:run_commands(?MODULE, Cmds),
+               stop(),
+               #state{distance = Distance, burnt = Burnt} = S,
+               Consumption = case Distance > 0 of
+                               true -> 100 * Burnt / Distance;
+                               false -> 0
+                             end,
+               ?WHENFAIL(
+                  io:format("Distance: ~p~nConsumption: ~p~n",
+                            [Distance, Consumption]),
+                  aggregate(command_names(Cmds),
+                            R =:= ok andalso (Distance < ?DISTANCE orelse
+                                              Consumption > ?CONSUMPTION)))
+             end)).
+
+%% This should fail most of the times with:<br/>
+%% `proper:quickcheck(car_fsm:prop_targeted_distance(), 1000).'
+prop_distance_targeted() ->
+  ?FORALL_TARGETED(
+     Cmds, more_commands(3, proper_fsm:targeted_commands(?MODULE)),
+     ?TRAPEXIT(
+        begin
+          start_link(),
+          {_H, {_, S}, R} = proper_fsm:run_commands(?MODULE, Cmds),
+          stop(),
+          #state{distance = Distance, burnt = Burnt} = S,
+          Consumption = case Distance > 0 of
+                          true -> 100 * Burnt / Distance;
+                          false -> 0
+                        end,
+          UV = Distance,
+          ?MAXIMIZE(UV),
+          ?WHENFAIL(
+             io:format("Distance: ~p~nConsumption: ~p~n",
+                       [Distance, Consumption]),
+             aggregate(command_names(Cmds),
+                       R =:= ok andalso (Distance < ?DISTANCE orelse
+                                         Consumption > ?CONSUMPTION)))
+        end)).
+
+%% This should fail most of the times with:<br/>
```

Comment:
Instead of this comment, better write here how this property differs from the one above and what its purpose is.